### PR TITLE
Add (de)activate to --help command list

### PR DESCRIPTION
**Pull request by 13rac1**
### Description

Add two mock entries to the configure_parser_init() so the `activate` and `deactivate` shell commands are displayed in the --help.

Before:
```
$ conda --help
...
commands:
  The following built-in and plugins subcommands are available.

  COMMAND
    clean             Remove unused packages and caches.
    compare           Compare packages between conda environments.
    config            Modify configuration values in .condarc.
    content-trust     See `conda content-trust --help`.
    create            Create a new conda environment from a list of specified packages.
    doctor            Display a health report for your environment.
    env               See `conda env --help`.
    info              Display information about current conda install.
    init              Initialize conda for shell interaction.
    install           Install a list of packages into a specified conda environment.
    list              List installed packages in a conda environment.
    notices           Retrieve latest channel notifications.
    pack              See `conda pack --help`.
    package           Create low-level conda packages. (EXPERIMENTAL)
    remove (uninstall)
                      Remove a list of packages from a specified conda environment.
    rename            Rename an existing environment.
    repoquery         Advanced search for repodata.
    run               Run an executable in a conda environment.
    search            Search for packages and display associated information using the MatchSpec format.
    server            See `conda server --help`.
    update (upgrade)  Update conda packages to the latest compatible version.
    verify            See `conda verify --help`.
```
After:
```
$ conda --help
...
commands:
  The following built-in and plugins subcommands are available.

  COMMAND
    activate          Activate a conda environment.
    clean             Remove unused packages and caches.
    compare           Compare packages between conda environments.
    config            Modify configuration values in .condarc.
    content-trust     See `conda content-trust --help`.
    create            Create a new conda environment from a list of specified packages.
    deactivate        Deactivate the current active conda environment.
    doctor            Display a health report for your environment.
    env               See `conda env --help`.
    info              Display information about current conda install.
    init              Initialize conda for shell interaction.
    install           Install a list of packages into a specified conda environment.
    list              List installed packages in a conda environment.
    notices           Retrieve latest channel notifications.
    pack              See `conda pack --help`.
    package           Create low-level conda packages. (EXPERIMENTAL)
    remove (uninstall)
                      Remove a list of packages from a specified conda environment.
    rename            Rename an existing environment.
    repoquery         Advanced search for repodata.
    run               Run an executable in a conda environment.
    search            Search for packages and display associated information using the MatchSpec format.
    server            See `conda server --help`.
    update (upgrade)  Update conda packages to the latest compatible version.
    verify            See `conda verify --help`.
```

This is arguably a hack, but is the correct way to resolve the issue according to #6353. I'm not 100% if they should be listed in `BUILTIN_COMMANDS`, but it seems right to me.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

**Comment by 13rac1** [comment]
FYI I have signed the CLA

**Comment by beeankha** [comment]
@conda-bot check

**Comment by beeankha** [comment]
@conda-bot check

File: news/13191-add-activate-to-help
Comment (beeankha) [inline news/13191-add-activate-to-help:3]:
```suggestion
* Add `activate` and `deactivate` to `--help` command list. (#13191)
```

Comment (13rac1) [reply news/13191-add-activate-to-help]: Done

**Review by beeankha — CHANGES_REQUESTED** [review]
Hi @13rac1 , thank you so much for creating this PR!

Something that needs to be addressed before this can be merged is that if someone has conda installed but `conda init` hasn't been run (you can run `$ unset conda` to recreate this scenario), then a very confusing error pops up and conda doesn't get properly activated:

```
> unset conda
> conda activate

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/Users/biancahenderson/Documents/GitHub/conda/conda/exception_handler.py", line 17, in __call__
        return func(*args, **kwargs)
      File "/Users/biancahenderson/Documents/GitHub/conda/conda/cli/main.py", line 83, in main_subshell
        exit_code = do_call(args, parser)
      File "/Users/biancahenderson/Documents/GitHub/conda/conda/cli/conda_argparse.py", line 159, in do_call
        relative_mod, func_name = args.func.rsplit(".", 1)
    AttributeError: 'Namespace' object has no attribute 'func'
```

In order to avoid this, a dummy/mock "execute" function for both `activate` and `deactivate` should be added, most likely in [`/conda/conda/cli/conda_argparse.py`](https://github.com/conda/conda/blob/main/conda/cli/conda_argparse.py), with instructions for the user to run `conda init` in order to improve the user experience and avoid possible confusion.

**Comment by 13rac1** [comment]
You're welcome! 

Haha, it's never easy is it?

These mock execute functions cannot go in `/conda/conda/cli/conda_argparse.py`, because the `do_call()` method expects to import a module in line 159: https://github.com/conda/conda/blob/f709c9655b413902a7ae093f8e35799dd3c1d69d/conda/cli/conda_argparse.py#L155-L164 

I could hack something into `main_init.py`, but it is best to just "do it right." I propose we add tiny `main_mock_activate.py` and `main_mock_deactivate.py` files, so `# func_name should always be 'execute' ` can remain correct. Is that a reasonable plan?

File: tests/test_activate.py
Comment (kenodegard) [inline tests/test_activate.py:3364]: We also want to test this with an environment:

```suggestion
    assert "ERROR: Run 'conda init' before 'conda activate'" in conda_cli("activate", "env")[0]
```

Comment (13rac1) [reply tests/test_activate.py:3364]: These messages will only appear in an unconfigured shell. They will not appear in a configured shell with an active environment. I may be missing something though, why do we want to test with an environment?

Comment (kenodegard) [reply tests/test_activate.py:3364]: Correct, however thats not what I'm getting at here.

If someone forgot to run `conda init` and tries to do `conda activate env` they will currently get a misleading error message instead of the useful message we want them to get:

```
$ unset conda  # or if using ZSH: `unset -f conda`
$ conda activate env
usage: conda [-h] [-v] [--no-plugins] [-V] COMMAND ...
conda: error: unrecognized arguments: env
```

**Comment by dholth** [comment]
Make this command return a nonzero shell error code.

**Comment by 13rac1** [comment]
@dholth That is a good point! You've phrased your suggestion as an order, please be mindful of your tone.

**Comment by dholth** [comment]
Fixes [#6353](https://github.com/conda/conda/issues/6353)

**Review by beeankha — APPROVED** [review]
Thank you for this contribution, @13rac1 !

**Comment by 13rac1** [comment]
You're welcome. Thanks! @beeankha @kenodegard @dholth

File: conda/cli/main_mock_activate.py
Comment (travishathaway) [inline conda/cli/main_mock_activate.py:11]:
```suggestion
from .. import CondaError


def execute(args, parser):
    raise CondaError("Run 'conda init' before 'conda activate'")
```

Let's raise a `CondaError` here instead. This has the following benefits:

- It's the standard way of showing errors to our users
- It automatically handles returning 1 as the exit code

File: conda/cli/main_mock_deactivate.py
Comment (travishathaway) [inline conda/cli/main_mock_deactivate.py:11]:
```suggestion
from .. import CondaError


def execute(args, parser):
    raise CondaError("Run 'conda init' before 'conda deactivate'")
```

Same thing.

File: tests/test_activate.py
Comment (travishathaway) [inline tests/test_activate.py:3374]:
```suggestion
    assert "Run 'conda init' before 'conda activate'" in conda_cli("activate")[0]
    assert (
        "Run 'conda init' before 'conda activate'"
        in conda_cli("activate", "env")[0]
    )

    # Call deactivate and check that the proper error shows up
    assert (
        "Run 'conda init' before 'conda deactivate'"
        in conda_cli("deactivate")[0]
    )
```

Updating tests.

**Review by travishathaway — CHANGES_REQUESTED** [review]
@13rac1,

Thanks for this pull request! I just have a couple of recommendations that I would like to see added before I give my approval. Please let me know if you have any questions.

**Comment by 13rac1** [comment]
@travishathaway Nope, makes sense. This is the difference in output:
```bash
# before
(base) test_user@f00a24b80e33:/opt/conda-src$ conda activate
ERROR: Run 'conda init' before 'conda activate'
(base) test_user@f00a24b80e33:/opt/conda-src$ 

#after
(base) test_user@f00a24b80e33:/opt/conda-src$ conda activate

CondaError: Run 'conda init' before 'conda activate'
(base) test_user@f00a24b80e33:/opt/conda-src$ 
```

File: tests/test_activate.py
Comment (13rac1) [reply tests/test_activate.py]: I've updated the tests to check for a `CondaError`:
```python
    with pytest.raises(CondaError) as conda_error:
       conda_cli("activate")
    assert conda_error.value.message.startswith("Run 'conda init' before 'conda activate'")
```

**Review by travishathaway — APPROVED** [review]

**Comment by 13rac1** [comment]
:shipit: Awesome! Thank you

**Pull request merged**
